crates/neon/src/types_impl/buffer/types.rs

```diff
@@ -0,0 +1,758 @@
+/// the storage of the underlying data buffer, and several typed views for managing access
+/// to the buffer. Neon provides access to the `ArrayBuffer` class with the
+/// [`JsArrayBuffer`](crate::types::JsArrayBuffer) type.
+///
+/// Node also provides a [`Buffer`][Buffer] type, which is built on top of `ArrayBuffer`
+/// and provides additional functionality. Neon provides access to the `Buffer` class
+/// with the [`JsBuffer`](crate::types::JsBuffer) type.
+///
+/// Many of Node's I/O APIs work with these types, and they can also be used for
+/// compact in-memory data structures, which can be shared efficiently between
+/// JavaScript and Rust without copying.
+///
+/// [u8]: std::primitive::u8
+/// [i8]: std::primitive::i8
+/// [u16]: std::primitive::u16
+/// [i16]: std::primitive::i16
+/// [u32]: std::primitive::u32
+/// [i32]: std::primitive::i32
+/// [u64]: std::primitive::u64
+/// [i64]: std::primitive::i64
+/// [f32]: std::primitive::f32
+/// [f64]: std::primitive::f64
+/// [JsUint8Array]: crate::types::JsUint8Array
+/// [JsInt8Array]: crate::types::JsInt8Array
+/// [JsUint16Array]: crate::types::JsUint16Array
+/// [JsInt16Array]: crate::types::JsInt16Array
+/// [JsUint32Array]: crate::types::JsUint32Array
+/// [JsInt32Array]: crate::types::JsInt32Array
+/// [JsBigUint64Array]: crate::types::JsBigUint64Array
+/// [JsBigInt64Array]: crate::types::JsBigInt64Array
+/// [JsFloat32Array]: crate::types::JsFloat32Array
+/// [JsFloat64Array]: crate::types::JsFloat64Array
+/// [Uint8Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array
+/// [Int8Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int8Array
+/// [Uint16Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint16Array
+/// [Int16Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int16Array
+/// [Uint32Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint32Array
+/// [Int32Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int32Array
+/// [BigUint64Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigUint64Array
+/// [BigInt64Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt64Array
+/// [Float32Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float32Array
+/// [Float64Array]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Float64Array
+/// [typed-arrays]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Typed_arrays
+/// [ArrayBuffer]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer
+/// [Buffer]: https://nodejs.org/api/buffer.html
+#[derive(Debug)]
+#[repr(transparent)]
+pub struct JsTypedArray<T: private::Binary> {
+    local: raw::Local,
```

Comment:
I noticed we're missing one more typed array feature. Neon doesn't expose a way to get the backing `ArrayBuffer`.

https://nodejs.org/api/n-api.html#napi_get_typedarray_info